api/envoy/config/filter/http/buffer/v2/buffer.proto

```diff
@@ -22,3 +22,15 @@ message Buffer {
   google.protobuf.Duration max_request_time = 2
       [(validate.rules).duration = {required: true, gt: {}}, (gogoproto.stdduration) = true];
 }
+
+message BufferPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    // Disable the buffer filter for this particular vhost or route.
+    bool disabled = 1 [(validate.rules).bool.const = true];
```

Comment:
I suggest you move this boolean into the filter itself. So that, your per route buffer config can reuse the main buffer filter message, and simply set disabled=true.

Reply:
I was debating that as a way of having it off by default and then enabled selectively per vhost/route. I like this idea but there'd be one challenge:

I'd still want a oneof to provide a clear delineation between disabled and enabled/configured. That would necessitate a breaking change though of the config which would be a no-go. @rshriram, I believe you're using an empty/zero config as the "disabled" mode in the fault filter. While I think that would work, I'd prefer the filter to be explicitly disabled by the config (instead of by convention), though that might be just a documentation change to clarify.

Thoughts?

Reply:
Yeah unfortunately I think it would be kind of ugly to backfill this into the main buffer config. IMO what you have is probably the best we can do.

Reply:
You can "upgrade" to oneof without breaking wire compat, so it's possible to do this in the core Buffer config.

Reply:
there are two fields currently in the config, we wouldn't be able to set both of them in the oneof unfortunately.